tests/qdwh_test.py

```diff
@@ -85,38 +89,45 @@ def testQdwhUnconvergedAfterMaxNumberIterations(
           'm': m, 'n': n, 'log_cond': log_cond}
       for m, n in zip([8, 10, 20], [6, 10, 18])
       for log_cond in np.linspace(1, _MAX_LOG_CONDITION_NUM, 4)))
-  # TODO(tianjianlu): Fails on A100 GPU and TPU.
-  @jtu.skip_on_devices("gpu", "tpu")
+  # TODO(tianjianlu): Fails on A100 GPU.
+  @jtu.skip_on_devices("gpu")
   def testQdwhWithUpperTriangularInputAllOnes(self, m, n, log_cond):
     """Tests qdwh with upper triangular input of all ones."""
-    a = jnp.triu(jnp.ones((m, n)))
+    a = jnp.triu(jnp.ones((m, n))).astype(_QDWH_TEST_DTYPE)
     u, s, v = jnp.linalg.svd(a, full_matrices=False)
     cond = 10**log_cond
     s = jnp.linspace(cond, 1, min(m, n))
     a = (u * s) @ v
     is_symmetric = _check_symmetry(a)
     max_iterations = 10
-    actual_u, actual_h, _, _ = qdwh.qdwh(a, is_symmetric, max_iterations)
+
+    with jax.default_matmul_precision('float32'):
```

Comment:
I would not add the scope around the call to qdwh. I would add the necessary precision overrides inside the qdwh implementation, since I'm not sure we have any evidence that QDWH works well with low matmul precision.